tests/integration/test_base_service.py

```diff
@@ -245,3 +261,188 @@ def test_get_service_interactions_raises(dispatcher: Dispatcher):
 
     with raises(ServiceNotFound):
         dispatcher.process_request(request_inexistent_service_interactions)
+
+
+@mark.parametrize(
+    'scenario_name, should_create_empty_dir',
+    [
+        ('test_scenario_record_creates_dir', False),
+        ('test_scenario_record_dir_exists', True),
+    ]
+)
+def test_base_service_process_response_record_creates_dir(
```

Comment:
This looks like they should be two different tests, adding the extra parameter to tweak the way the setup of the test is done is a bit confusing, either split or improve the naming of it :/